pkg/readiness/health/health.go

```diff
@@ -30,7 +30,7 @@ type processHealth struct {
 	IsInGoalState   bool               `json:"IsInGoalState"`
 	LastMongoUpTime int64              `json:"LastMongoUpTime"`
 	ExpectedToBeUp  bool               `json:"ExpectedToBeUp"`
-	ReplicaStatus   *replicationStatus `json:"ReplicationStatus,omitempty"`
```

Comment:
I'm concerned about this change, (please correct me if I'm wrong).

In older versions of the agent this attribute won't be available, what will happen in that case. It is possible to use this readinessprobe version with an older version of the agent (not community, but Enterprise), if deploying an old version of Ops Manager, for instance. In that case `ReplicationStatus` won't be there.

Reply:
I believe that in that case `ReplicaStatus` will just be `nil`?

I also tried to do the following change:

```
diff --git a/cmd/readiness/testdata/health-status-ok-no-replica-status.json b/cmd/readiness/testdata/health-status-ok-no-replica-status.json
index caaf67f..0398e8e 100644
--- a/cmd/readiness/testdata/health-status-ok-no-replica-status.json
+++ b/cmd/readiness/testdata/health-status-ok-no-replica-status.json
@@ -73,7 +73,6 @@
     },
     "statuses": {
         "bar": {
-            "ReplicationStatus": null,
             "ExpectedToBeUp": true,
             "LastMongoUpTime": 1568222195,
             "IsInGoalState": true
```

And to run the following unit test:

```
	t.Run("MongoDB replication state is reported by agents", func(t *testing.T) {
		assert.True(t, isPodReady(testConfig("testdata/health-status-ok-no-replica-status.json")))
	})
```

And it works fine. Does this answer the question? I can add an explicit new unit test to cover this

Reply:
It does not hurt to add that test, let's cover this with a unit test.